tls/s2n_connection.c

```diff
@@ -554,10 +467,21 @@ int s2n_connection_wipe(struct s2n_connection *conn)
     }
     POSIX_GUARD_RESULT(s2n_prf_wipe(conn));
     struct s2n_prf_working_space *prf_workspace = conn->prf_space;
+    if (!conn->initial) {
+        POSIX_GUARD_RESULT(s2n_crypto_parameters_new(&conn->initial));
+    } else {
+        POSIX_GUARD_RESULT(s2n_crypto_parameters_wipe(conn->initial));
+    }
+    struct s2n_crypto_parameters *initial = conn->initial;
+    if (!conn->secure) {
+        POSIX_GUARD_RESULT(s2n_crypto_parameters_new(&conn->secure));
+    } else {
+        POSIX_GUARD_RESULT(s2n_crypto_parameters_wipe(conn->secure));
+    }
+    struct s2n_crypto_parameters *secure = conn->secure;
```

Comment:
What's happening here exactly? Why save a pointer to conn->secure/conn->initial?

Reply:
The first half of s2n_connection_wipe is saving anything from the connection we want to keep, then we zero the connection structure, then we restore the things we saved. The stack pointers here are to save the conn->secure/conn->initial memory. Without them, the conn->secure and conn->initial pointers will be reset to zero/NULL and the memory they pointed to will be leaked.